source/server/listener_impl.h

```diff
@@ -330,6 +335,12 @@ class ListenerImpl final : public Network::ListenerConfig,
   void createUdpListenerFilterChain(Network::UdpListenerFilterManager& udp_listener,
                                     Network::UdpReadFilterCallbacks& callbacks) override;
 
+#ifdef ENVOY_ENABLE_QUIC
+  void setQuicStatNames(Quic::QuicStatNames& quic_stat_names) {
```

Comment:
Rather than having this OptRef flow can we set this in the constructor from the ListenerManagerFactory that's passed in?